runtime/compiler/trj9/control/CompilationThread.cpp

```diff
@@ -6136,58 +6136,38 @@ TR::CompilationInfoPerThreadBase::generatePerfToolEntry()
 #endif
    }
 
-
-// The compilation thread has VM access when it enters this method
-// wrappedCompile() may release VM access
-void *
-TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
-                                         TR_MethodToBeCompiled *entry,
-                                         J9::J9SegmentProvider &scratchSegmentProvider)
-   {
-   {
-   PORT_ACCESS_FROM_JITCONFIG(jitConfig);
-   setTimeWhenCompStarted(j9time_usec_clock());
-   }
-   J9JavaVM *javaVM = vmThread->javaVM;
-   TR::RawAllocator rawAllocator(javaVM);
-   J9::SystemSegmentProvider defaultSegmentProvider(
-      1 << 16,
-      (0 != scratchSegmentProvider.getPreferredSegmentSize()) ? scratchSegmentProvider.getPreferredSegmentSize()
-                                                              : 1 << 24,
-      TR::Options::getScratchSpaceLimit(),
-      scratchSegmentProvider,
-      rawAllocator
-      );
-   TR::DebugSegmentProvider debugSegmentProvider(
-      1 << 16,
-      rawAllocator
-      );
-   TR::SegmentAllocator &regionSegmentProvider =
-      TR::Options::getCmdLineOptions()->getOption(TR_EnableScratchMemoryDebugging) ?
-         static_cast<TR::SegmentAllocator &>(debugSegmentProvider) :
-         static_cast<TR::SegmentAllocator &>(defaultSegmentProvider);
-   TR::Region dispatchRegion(regionSegmentProvider, rawAllocator);
-   TR_Memory trMemory(*_compInfo.persistentMemory(), dispatchRegion);
-   TR_J9VMBase * vm = NULL;
-   _qszWhenCompStarted = getCompilationInfo()->getMethodQueueSize();
-   TR_ASSERT(!_compiler, "The previous compilation was not properly cleared.");
-
-   //start new compilation
-   TR_MethodMetaData *metaData = 0;
-   void *startPC = 0;
-   J9Method *method = entry->getMethodDetails().getMethod();
-
-   vmThread->jitMethodToBeCompiled = method;
-
+/**
+ * @brief TR::CompilationInfoPerThreadBase::preCompilationTasks
+ * @param vmThread Pointer to the current J9VMThread
+ * @param javaVM Pointer to the J9JavaVM
+ * @param entry Pointer to the method to be compiled entry
+ * @param method Pointer to the J9Method
+ * @param aotCachedMethod Pointer to a pointer to the AOT code to be relocated (if it exists)
+ * @param trMemory Reference to the TR_Memory associated with this compilation
+ * @param canDoRelocatableCompile Reference to a bool; indicates if the current compilation can be AOT compiled
+ * @param eligibleForRelocatableCompile Reference to a bool; Indicates if the current compilation is eligible for AOT
+ * @param reloRuntime Pointer to a TR_RelocationRuntime; NULL if JVM does not have AOT support
```

Comment:
I would like to specify which parameters are for input and which are for output. For instance `canDoRelocatableCompile` is an output parameter set the function itself.
Input parameters should be marked 'const' as much as possible.